src/client/src/containers/AppServiceModal/index.tsx

```diff
@@ -0,0 +1,617 @@
+/**
+ * This component uses React Hooks in lieu of Class Components.
+ * https://reactjs.org/docs/hooks-intro.html
+ */
+import * as React from "react";
+import { connect } from "react-redux";
+
+import Dropdown from "../../components/Dropdown";
+import asModal from "../../components/Modal";
+
+import { closeModalAction } from "../../actions/modalActions/modalActions";
+import { saveAppServiceSettingsAction } from "../../actions/azureActions/appServiceActions";
+import { appServiceModalInitialState } from "../../mockData/cosmosDbModalData";
+import { azureMessages as azureModalMessages } from "../../mockData/azureServiceOptions";
+import { messages } from "./messages";
+import { ReactComponent as Spinner } from "../../assets/spinner.svg";
+import { ReactComponent as Cancel } from "../../assets/cancel.svg";
+import { ReactComponent as GreenCheck } from "../../assets/checkgreen.svg";
+import { getAppServiceSelectionInDropdownForm } from "../../selectors/appServiceSelector";
+import { isAppServiceModalOpenSelector } from "../../selectors/modalSelector";
+
+import { InjectedIntlProps, injectIntl } from "react-intl";
+
+import { setAppServiceModalButtonStatus } from "./verifyButtonStatus";
+
+import buttonStyles from "../../css/buttonStyles.module.css";
+import {
+  EXTENSION_COMMANDS,
+  EXTENSION_MODULES,
+  WIZARD_CONTENT_INTERNAL_NAMES,
+  KEY_EVENTS
+} from "../../utils/constants";
+import styles from "./styles.module.css";
+import { Dispatch } from "redux";
+import { setAzureValidationStatusAction } from "../../actions/azureActions/setAzureValidationStatusAction";
+import {
+  setSiteNameAvailabilityAction,
+  IAvailabilityFromExtension
+} from "../../actions/azureActions/setAccountAvailability";
+import { AppState } from "../../reducers";
+import { getVSCodeApiSelector } from "../../selectors/vscodeApiSelector";
+import RootAction from "../../actions/ActionType";
+import { ISelected } from "../../types/selected";
+import { ISelectionInformation } from "../../selectors/appServiceSelector";
+import { IAvailability } from "../../reducers/wizardSelectionReducers/services/appServiceReducer";
+import { IVSCodeObject } from "../../reducers/vscodeApiReducer";
+import { ISubscriptionData } from "../../reducers/azureLoginReducers/subscriptionDataReducer";
+import classNames from "classnames";
+
+const DEFAULT_VALUE = {
+  value: "Select...",
+  label: "Select..."
+};
+
+interface IStateProps {
+  isModalOpen: boolean;
+  vscode: IVSCodeObject;
+  subscriptionData: ISubscriptionData;
+  subscriptions: [];
+  isValidatingName: boolean;
+  siteNameAvailability: IAvailability;
+  selection: any;
```

Comment:
Discussed this with @njindal239 and could not figure out a good, simple way to type this (had trouble getting it working with `getAppServiceSelection`). Currently this is also `any` for this for Cosmos and Functions. So it might be better to type this after doing some refactoring for all the forms. this If you guys have suggestions let me know.

Reply:
what is the variable "selection" for?

Reply:
This is typed now.

`selection` stores the form data in the appService store